WordPress/Classes/Utility/ZendeskUtils.swift

```diff
@@ -37,78 +36,38 @@ import CoreTelephony
                                         clientId: clientId)
 
         ZendeskUtils.toggleZendesk(enabled: true)
-        ZendeskUtils.createIdentity()
     }
 
     func showHelpCenter(from controller: UIViewController) {
+        ZendeskUtils.createIdentity(success: {
 
-        if !identityCreated {
-            return
-        }
-
-        guard let helpCenterContentModel = ZDKHelpCenterOverviewContentModel.defaultContent() else {
-            return
-        }
+            guard let helpCenterContentModel = ZDKHelpCenterOverviewContentModel.defaultContent() else {
+                DDLogInfo("Zendesk helpCenterContentModel creation failed.")
+                return
+            }
 
-        helpCenterContentModel.groupType = .category
-        helpCenterContentModel.groupIds = [Constants.mobileCategoryID]
-        helpCenterContentModel.labels = [Constants.articleLabel]
+            helpCenterContentModel.groupType = .category
+            helpCenterContentModel.groupIds = [Constants.mobileCategoryID]
+            helpCenterContentModel.labels = [Constants.articleLabel]
 
-        let presentInController = ZendeskUtils.configureViewController(controller)
-        ZDKHelpCenter.presentOverview(presentInController, with: helpCenterContentModel)
+            let presentInController = ZendeskUtils.configureViewController(controller)
+            ZDKHelpCenter.presentOverview(presentInController, with: helpCenterContentModel)
+        }, failure: nil)
```

Comment:
Few thoughts on alternatives!!

```
func showHelpCenterIfPossible(from controller:  UIViewController) {
    ZendeskUtils.createIdentity { success in {
          guard success else { 
              showFailureAlert(from: controller)
              return
          }

          showHelpCenter(from: controller)
    }
}

private func showHelpCenter(from controller: UIViewController) {
     // All of the `showHelpCenter` method's code, minus the Identity creation
}
```

- The main idea is to let the caller know that the method **may fail** (instead of silently failing).
- By updating the `createIdentify` signature to `private func createIdentity(completion: (Bool) -> Void)` you can *slightly* simplify the syntax. (Arguable point, definitely!!)
- This applies to the other public methods as well!

Reply:
Thanks @jleandroperez . I'll look at changing it. I hadn't quite figured out how I wanted to handle failures, but I can at least structure it well so it'll be easy to do/change in the future.